tests/test_layout.py

```diff
@@ -735,3 +738,78 @@ def test_tree_style_utf8(hlwm):
       … ├─╼ vertical: [FOCUS]
       … ╰─╼ vertical:
     """)
+
+
+def test_split_invalid_argument(hlwm):
+    # this also tests all exceptions in Converter<FixPrecDec>::parse()
+    wrongDecimal = [
+        ('0.0f', "After '.' only digits"),
+        ('0.+8', "After '.' only digits"),
+        ('0.-8', "After '.' only digits"),
+        ('0..0', "A decimal must have at most one '.'"),
+        ('.3', "There must be at least one digit"),
+        ('.', "There must be at least one digit"),
+        ('b', "stoi"),
+        ('-.3', "stoi"),
+        ('+.8', "stoi"),
+    ]
+    for d, msg in wrongDecimal:
```

Comment:
Why those for loops rather than `@pytest.mark.parametrize` (which will run the tests isolated from each other and don't e.g. abort the remaining cases if one fails)?

Reply:
Let me continue the discussion here. :-)

> For performance reasons. 

That's an understandable reason. @The-Compiler I think we should take this as input to explore if we can achieve this runtime behavior without having it bleed into the test definitions like this.

Reply:
Hmm, I see! I guess we could have a `hlwm_shared` fixture (or whatever) with `scope='module'`, but that'd then be the same hlwm process for the entire test module. Also see: https://github.com/pytest-dev/pytest/issues/1552

Reply:
Isn't this makeing the testsuite more complicated than it needs to be? Or asking the other way round: where is the issue with the test case as it is at the moment?

Reply:
IMHO it's not *much* of an issue, but I don't think it is desirable that you have to take these things into account when writing tests. This should be abstracted away.

Just to be clear: I'm quite fine with how this looks right now, but I'd be happier if the test framework removed the need for you to make that choice (and get the same performance advantage).

An ideal outcome would be that there is much less restarting of hlwm happening in a full test run, because I think a lot of tests do not strictly need a fresh instance.